src/Nethermind/Nethermind.Merge.Plugin/Synchronization/MergeBlockDownloader.cs

```diff
@@ -110,33 +108,54 @@ public class MergeBlockDownloader : BlockDownloader
 
             bool HasMoreToSync()
                 => currentNumber < _blockTree.BestKnownBeaconNumber &&
-                   bestPeer.HeadNumber > _blockTree.BestKnownNumber;
+                   bestPeer.HeadNumber > currentNumber;
+
+            bool foundBeaconBlocks = false;
 
             while (HasMoreToSync())
             {
                 if (_logger.IsDebug)
                     _logger.Debug($"Continue full sync with {bestPeer} (our best {_blockTree.BestKnownNumber})");
+
+                long upperDownloadBoundary = _blockTree.BestKnownBeaconNumber;
```

Comment:
I deliberately removed blocksLeft and upperDownloadBoundary. Do you think we should revert it?

Reply:
this is left over from rebase from master, will refactor